src/react.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useCallback, useContext, useMemo, useRef } from "react";
+import { WTEventParams } from "./client";
+
+type ContextValue<Params = Record<string, any>> = {
+  params: React.RefObject<Partial<Params>>;
+  track: (overrides?: Partial<Params>) => void;
+};
+
+type Options<Params> = {
+  validateEvent?: (params: Partial<Params>) => boolean;
+};
+
+export const createProvider = <
+  Params extends Record<string | number, any> = WTEventParams
```

Comment:
do we want to allow number keys?

Reply:
i don't think there's a particular reason to exclude them?

Reply:
curious what a practical use case would be, but don't feel strongly about excluding them.